### DISCUSSION TITLE
How to integrate libraries like NextUI or TanStack Query which wrap providers?

### DISCUSSION THREAD

**Discussion by neeraj-codebuddy** [web-infra-dev/modern.js#4880] [Q&A]
Hi, I recently started exploring new frameworks and came across Modern.js. I'm liking it very much and wanted to play around with it a bit. While doing so, I also wanted to add [NextUI which requires it be wrapped around a provider](https://nextui.org/docs/guide/installation#provider-setup). However, I did not find any guide on how to include libraries which require wrapping your application `Provider`. Same goes for [TanStack Query](https://tanstack.com/query/latest/docs/react/quick-start) as well. Any help is appreciated.

**Comment by guangzan** [comment]
same question

**Comment by yimingjfe** [comment]
Hey, I don't know much about NextUI, but you can add a Provider in `routes/layout.tsx` like this: 
```tsx
export default function layout(){
  return (
    <Provider>
      <Component1>
        <Outlet />
      </Component1>
    </Provider>
  )
}
```
